From dc48989ae9f332f9b12978dbbd3f02b353ea4479 Mon Sep 17 00:00:00 2001
From: Benson Leung <bleung@chromium.org>
Date: Wed, 4 Apr 2012 17:26:47 -0700
Subject: [PATCH 10/57] CHROMIUM: Input: atmel_mxt_ts - change MXT_BOOT_LOW to
 0x26

Changed to support address on new part. This is contrary to
Atmel documentation, and we are still working with them
to clear up this inconsistency. In the meantime,
change it for now to allow existing systems to update.

BUG=chrome-os-partner:8734
TEST=Firmware update works.

Change-Id: Ib2db2a066126df291c7a85208743c80c5357d55d
Signed-off-by: Benson Leung <bleung@chromium.org>
Reviewed-on: https://gerrit.chromium.org/gerrit/19639
Reviewed-by: Daniel Kurtz <djkurtz@chromium.org>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index ef867d3..0aae3fb 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -30,7 +30,11 @@
 /* Slave addresses */
 #define MXT_APP_LOW		0x4a
 #define MXT_APP_HIGH		0x4b
-#define MXT_BOOT_LOW		0x24
+/*
+ * MXT_BOOT_LOW disagrees with Atmel documentation, but has been
+ * updated to support new touch hardware that pairs 0x26 boot with 0x4a app.
+ */
+#define MXT_BOOT_LOW		0x26
 #define MXT_BOOT_HIGH		0x25
 
 /* Firmware */
-- 
1.8.1.4

